### ISSUE TITLE
Issue with changing password validation rule

### ISSUE THREAD

**Issue by minjup** [FusionAuth/fusionauth-netcore-client#12]
Version that we are using 1.10.0 for Fusion Auth .net core client and 1.10.1 for FusionAuth 

We've created a tenant with password validation rule of one special character and then changed the password validation rule.

![image](https://user-images.githubusercontent.com/29390587/74614698-04b6b280-517f-11ea-879d-3aca705311af.png)

When we change the password validation rule in FusionAuth UI (to enable more rules) for that tenant, it works fine. However, `RetrieveTenant` method in the .net core client returns status code 200 with error response.

Below is the error response that we get
```
tenant.eventConfiguration.events['user.delete'] Could not convert string 'user.delete' to dictionary key type 'io.fusionauth.domain.event.EventType'. Create a TypeConverter to convert from the string to the key type object. Path 'tenant.eventConfiguration.events['user.delete']', line 1, position 557.
```

I've tried to retrieve tenant with Postman and this works fine.

**Comment by matthew-altman** [comment]
@minjup Can you please try retesting with the NETCore client version 1.10.1, so that it matches your FusionAuth version of 1.10.1? There were some client domain object changes between those versions.
Please reply back with the results of that test. Thank you!

**Comment by minjup** [comment]
@matthew-altman the reason why we are using NETCore client version 1.10.0 is because there is no 1.10.1 available in nuget manager :D the next one available after 1.10.0 is 1.11.0

**Comment by matthew-altman** [comment]
@minjup My apologies. Even though we have a tagged 1.10.1 version of the netcore client in github, we don't always push patch versions to nuget.
After further research, it looks like the actual fix didn't make it into the client until 1.11.0.
I was able to use this version of the client and write some tests to validate your specific scenario of creating and retrieving a tenant with the eventConfiguration, and confirmed that the json unmarshaling error no longer exists.
Would it be possible to use the 1.11.0 client? Should still be fine on your 1.10.1 instance of FA.
